nemo_skills/inference/server/model.py

```diff
@@ -224,6 +236,53 @@ def __call__(
             )
         return outputs
 
+    def __recover_from_error(self, request, new_output, executor):
+        recovery_request = {key: value for key, value in request.items() if key != 'prompts'}
+        recovery_request['prompts'] = [new_output['full_prompt']]
+
+        # if greedy, setting parameters to random
+        if request['temperature'] == 0 or request['top_k'] == 1:
+            recovery_request['temperature'] = 0.7
```

Comment:
Can we take this as a new data class called error_recovery and then you can add attempts as well as the temperature and other parameters there. + an option to enable/disable majority voting, so that it all can be customized?